widget/accordioncontainer.go

```diff
@@ -0,0 +1,335 @@
+
+type accordionItem struct {
+	BaseWidget
+	container *AccordionContainer
+	header    *accordionItemHeader
+	detail    fyne.CanvasObject
+	open      bool
+}
+
+func (i *accordionItem) setOpen(open bool) {
+	if i.open == open {
+		return
+	}
+	if i.open = open; i.open {
+		i.detail.Show()
+	} else {
+		i.detail.Hide()
+	}
+}
+
+// CreateRenderer is a private method to Fyne which links this widget to its renderer
+func (i *accordionItem) CreateRenderer() fyne.WidgetRenderer {
+	return &accordionItemRenderer{
+		item: i,
+		objects: []fyne.CanvasObject{
+			i.header,
+			i.detail,
+		},
+	}
+}
+
+type accordionItemRenderer struct {
+	item    *accordionItem
+	objects []fyne.CanvasObject
+}
+
+func (r *accordionItemRenderer) MinSize() fyne.Size {
+	min := r.item.header.MinSize()
+	width := min.Width
+	height := min.Height
+	if r.item.open {
+		min := r.item.detail.MinSize()
+		width = fyne.Max(width, min.Width)
+		height += min.Height
+	}
+	return fyne.NewSize(width, height)
+}
+
+func (r *accordionItemRenderer) Layout(size fyne.Size) {
+	height := r.item.header.MinSize().Height
+	r.item.header.Move(fyne.NewPos(0, 0))
+	r.item.header.Resize(fyne.NewSize(size.Width, height))
+	if r.item.open {
+		r.item.detail.Move(fyne.NewPos(0, height))
+		r.item.detail.Resize(fyne.NewSize(size.Width, r.item.detail.MinSize().Height))
+	}
+}
+
+func (r *accordionItemRenderer) BackgroundColor() color.Color {
+	return theme.BackgroundColor()
+}
+
+func (r *accordionItemRenderer) Objects() []fyne.CanvasObject {
+	return r.objects
+}
+
+func (r *accordionItemRenderer) Refresh() {
+	r.item.header.Refresh()
+	r.item.detail.Refresh()
+	canvas.Refresh(r.item)
+}
+
+func (r *accordionItemRenderer) Destroy() {
+}
+
+var _ fyne.Tappable = (*accordionItemHeader)(nil)
+var _ fyne.Widget = (*accordionItemHeader)(nil)
+var _ desktop.Hoverable = (*accordionItemHeader)(nil)
+
+type accordionItemHeader struct {
+	BaseWidget
+	item    *accordionItem
+	text    string
+	hovered bool
+}
+
+// Tapped is called when a pointer tapped event is captured and triggers any tap handler
+func (h *accordionItemHeader) Tapped(*fyne.PointEvent) {
+	h.item.setOpen(!h.item.open)
+	h.item.container.Refresh()
+}
+
+// MouseIn is called when a desktop pointer enters the widget
+func (h *accordionItemHeader) MouseIn(*desktop.MouseEvent) {
+	h.hovered = true
+	h.item.container.Refresh()
+}
+
+// MouseOut is called when a desktop pointer exits the widget
+func (h *accordionItemHeader) MouseOut() {
+	h.hovered = false
+	h.item.container.Refresh()
+}
+
+// MouseMoved is called when a desktop pointer hovers over the widget
+func (h *accordionItemHeader) MouseMoved(*desktop.MouseEvent) {
+}
+
+// CreateRenderer is a private method to Fyne which links this widget to its renderer
+func (h *accordionItemHeader) CreateRenderer() fyne.WidgetRenderer {
+	r := &accordionItemHeaderRenderer{
+		header: h,
+		images: []*canvas.Image{
+			canvas.NewImageFromResource(theme.ContentRemoveIcon()),
```

Comment:
Could it use arrows instead? As the names suggest this is for add and remove of content

Reply:
Yup, switched to MoveUp/MoveDown Icons

Reply:
Note: this is a deviation from #206 which requested '+'